pkgs/development/python-modules/llmx/default.nix

```
@@ -0,0 +1,71 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  setuptools-scm,
+  setuptools,
+  pythonOlder,
+  pydantic,
+  openai,
+  tiktoken,
+  diskcache,
+  cohere,
+  google-auth,
+  typer,
+  pyyaml,
+  transformers,
+  fastapi,
+  uvicorn,
+  accelerate,
+  withWeb ? true,
+  withTransformers ? true,
+}:
+
+buildPythonPackage rec {
+  pname = "llmx";
+  version = "0.0.21a0";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-OEo6wIaDTktzAsP0rOmhxjFSHygTR/EpcRI6AXsu+6M=";
+  };
+
+  disabled = pythonOlder "3.9";
+
+  build-system = [ setuptools-scm ];
+
+  dependencies =
+    [
+      pydantic
+      openai
+      tiktoken
+      diskcache
+      cohere
+      google-auth
+      typer
+      pyyaml
+      setuptools
+    ]
+    ++ lib.optionals withWeb [
+      fastapi
+      uvicorn
+    ]
+    ++ lib.optionals withTransformers [
+      transformers
+      accelerate
+    ];
+
+  # Tests of llmx try to access openai, google, etc.
+  # nativeCheckInputs = [pytestCheckHook];
```

This makes sure that no tests are executed.

```suggestion
  doCheck = false;
```